src/librbd/io/ImageRequestWQ.cc

```diff
@@ -218,6 +218,8 @@ void ImageRequestWQ<I>::aio_read(AioCompletion *c, uint64_t off, uint64_t len,
     return;
   }
 
+  m_image_ctx.read_iops_throttle->get(1);
```

Comment:
Prefer not to block the dispatch thread -- the other PR already has a solution to just not pop IO off the work queue. With some tweaking, it could support independent read and write throttling.